backend/deepchecks_monitoring/logic/check_logic.py

```diff
@@ -694,3 +697,60 @@ def reduce_check_window(model_results, monitor_options):
         if it is not None else None
         for model_version, it in model_results.items()
     }
+
+
+async def get_query_with_cache(
```

Comment:
also, is it a good idea, in our case, to store big payloads in the Redis?

quick answer from stackoverflow (_not sure how legit/correct it is_) - [link](https://stackoverflow.com/a/55523632)

maybe materialized views can be used?

Reply:
there isn't a definite answer on SO
The problem is in the analysis we are doing the same query 8 times (for each check), and we would like to solve that somehow. The Redis option is an option we talked about trying, but we need to think more thoroughly about whether it will help in a real-life scenarios and not just in stress tests